src/Billing/BillingProcessor/Tasks/GeneratorX12.php

```diff
@@ -95,6 +95,18 @@ public function setup(array $context)
      */
     public function validateOnly(BillingClaim $claim)
     {
+        $return = BillingUtilities::updateClaim(
+            true,
```

Comment:
hi @kchapple , don't have to add a record to the claims table with validate only.

Reply:
Thanks @stephenwaite I didn't think so but Dan @growlingflea added this to the New Leaf repo, he said it was required for the validation to display all the information.
Ken

Reply:
Hi @kchapple, not able to reproduce the issue so we should probably start an issue with more detail as to the problem.

Reply:
Dan says "it has to be a new claim.  And it has to be 'validate only'  Once the claim is created, this doesn't happen"

Reply:
We're working on reproducing again

Reply:
Dan:

> Let me reword this:  If Validate Only is selected before the X12 file is created, the ins. info show blank.  If the end user selects "Continue" then reopens it and selects "Validate" the ins. info is there.

Reply:
I'm still not convinced that updating the billing table is the right thing to do.

Reply:
Here are explicit steps to reproduce:

[Missing Ins. Info Bug.pdf](https://github.com/openemr/openemr/files/8181213/Missing.Ins.Info.Bug.pdf)

Reply:
We think that nothing is changed in the DB since status and bill_process stay the same.

Reply:
For #4979, the payer id isn't getting added to the row created in the billing table from the fee sheet. This can be addressed there or can add a condition to the `loadPayerInfo()` function to check for a null payer id on [this line](https://github.com/openemr/openemr/blob/master/src/Billing/Claim.php#L241).
```
if (($drow['provider'] == $billrow['payer_id'] && empty($this->payers[0]['data'])) || $billrow['payer_id'] == null) {
```

Reply:
actually should probably be
```
if (($drow['provider'] == $billrow['payer_id']  || $billrow['payer_id'] == null) && empty($this->payers[0]['data'])) { {